src/lib/utils.ts

```diff
@@ -191,7 +193,19 @@ export const toFixed = (num: number | string, decimalPlaces = 8) => {
   return n.toFixed(decimalPlaces).replace(/\.?0+$/, '')
 }
 
-export const SHARED_MODAL_BOX_SHADOW_STYLE = '0px 0px 16.4px 0px rgba(229,107,26,0.68)'
+/**
+ * Formats a number with commas
+ * @param num - The number to format
+ * @returns The formatted number with commas
+ * @example formatNumberWithCommas(123456789) // '123,456,789'
+ * @example formatNumberWithCommas(1234567.89) // '1,234,567.89'
+ * @example formatNumberWithCommas(0.000123) // '0.000123'
+ */
+export function formatNumberWithCommas(num: number | string | bigint): string {
+  const parts = num.toString().split('.')
```

Comment:
We need a NaN validation here, like toFixed

Reply:
Done